### DISCUSSION TITLE
Is it possible to get a list of all the variables that will be rendered in the output template?

### DISCUSSION THREAD

**Discussion by samsmith123** [harttle/liquidjs#496] [Q&A] [answered]
I would like to get a list of all the variables that are used in the template, it could be a simple output or in a tag, also if the variable is in a block that won't be rendered, it shouldn't be in that list, for example, if the variable is used in an output of an if condition block, and the if condition is false, that variable isn't being rendered anymore so it shouldn't be in the list if it is not being used anywhere else.
Thanks.

**Comment by harttle** [comment, marked as answer]
I guess you can achieve this by recursively converting the scope passed into Liquid to getter/setter, via [defineProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#syntax).

Currently LiquidJS doesn't and has no plan to provide hooks for context read.
